### ISSUE TITLE
WASM decoder

### ISSUE THREAD

**Issue by wipfli** [maplibre/maplibre-tile-spec#225]
Could we use a decoder written in C++ or Rust and deploy it as webassembly? Could this give performance benefits?

**Comment by mactrem** [comment]
Yes, this is one of my goals to write a Rust decoder and compile it to WebAssembly. Only with WASM we can use SIMD instructions in the browser and take full advantage of the encodings of the format

**Comment by CommanderStorm** [comment]
Implemented in https://github.com/maplibre/maplibre-tile-spec/pull/1062

Needs lots of optimisation work though, currently not actually better
